src/server/scripts/Spells/spell_hunter.cpp

```diff
@@ -480,7 +480,8 @@ class spell_hun_misdirection : public SpellScriptLoader
 
             void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
             {
-                if (GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_DEFAULT)
+                if (GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_DEFAULT ||
+                   (GetTargetApplication()->GetRemoveMode() == AURA_REMOVE_BY_DEFAULT && !GetTarget()->HasAura(SPELL_HUNTER_MISDIRECTION_PROC)))
```

Comment:
"GetTargetApplication()->GetRemoveMode() == AURA_REMOVE_BY_DEFAULT" part is unneeded, it's already implicit by the previous "GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_DEFAULT" check and the || operator